src/compile/data/bin.ts

```diff
@@ -89,6 +90,38 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeTransform(model: ModelWithField, t: BinTransform) {
```

Comment:
As mentioned in person, you can name this `makeBinFromTransform` and rename the other make to `makeBinFromEncoding`.